compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/PackageLoader.java

```diff
@@ -319,55 +319,64 @@ private Resolution resolveModuleByPath(PackageID moduleID) {
      * 2. If a version is available on the Ballerina.toml
      * 3. If a version is available in the Ballerina.toml of the enclosing module's balo.
      *
-     * @param moduleID The ID of the module.
+     * @param moduleID      The ID of the module.
      * @param enclPackageId The ID of the parent module.
      */
     private void updateModuleIDVersion(PackageID moduleID, PackageID enclPackageId) {
         String orgName = moduleID.orgName.value;
         String moduleName = moduleID.name.value;
-        
+        String moduleVersion;
+
         // Set the version from the Ballerina.lock file found in the current project.
-        if (enclPackageId != null && moduleID.version.value.isEmpty() &&
-            this.hasLockFile(Paths.get(this.options.get(PROJECT_DIR)))) {
+        if (enclPackageId != null && this.hasLockFile(Paths.get(this.options.get(PROJECT_DIR)))) {
             // Not a top level package or bal
             if (this.lockFile.getImports().containsKey(enclPackageId.toString())) {
                 List<LockFileImport> foundBaseImport = lockFile.getImports().get(enclPackageId.toString());
                 Optional<LockFileImport> foundNestedImport = foundBaseImport
-                                                                      .stream()
-                                                                      .filter(nestedImport ->
-                                                      moduleID.orgName.value.equals(nestedImport.getOrgName()) &&
-                                                      moduleID.name.value.equals(nestedImport.getName()))
-                                                                      .findFirst();
-                foundNestedImport.ifPresent(dependencyPkg -> moduleID.version =
-                        new Name(dependencyPkg.getVersion()));
+                        .stream()
```

Comment:
I think something like below is more simple to read

```java
for (LockFileImport nestedImport : foundBaseImport) {
    if (moduleID.orgName.value.equals(nestedImport.getOrgName()) &&
                                        moduleID.name.value.equals(nestedImport.getName())) {
        moduleID.version = new Name(nestedImport.getVersion());
        return;
    }
}
moduleID.version = new Name("");

```

Also applicable to L:352

Reply:
Fixed in fa4cab4